.github/workflows/check.yml

```diff
@@ -0,0 +1,26 @@
+# Reference: https://github.com/golangci/golangci-lint-action
+name: Check
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+
+permissions:
```

Comment:
For what is that needed? 🤔

Reply:
I think by default there are lot more permissions given to the job but here we are actually asking explicitly only what we require. Again not 100% sure...not an expert in github actions I am a gitlab guy :wink: 

I just took the reference example as shared by golangci.